Merge generate_code_and_verify_string to dev

Vincentmask opened the pull request:
1. Removed the first three import since we can pass in the client on the server directly.
2. Changing var to let  & const to provide block scoping for variables.
3. In the function that require to connect to the db, added client to the params, so we don't need to connect to db everytime.
4. The sortChars function returns a comma-separated string instead of an array. Therefore, when the concat method is called on the arr variable, it does not modify the array but instead returns a new array with the concatenated value. The concat and return arr.sort().toString() in function 'sortChars' is returning a list of 'undefine' So I changed it to using push, and return arr.sort().join('');
5. Modified queries to using values to prevent malicious actions.
6. use += to replace concat since the .concat() method is used to concatenate arrays or strings, and it returns a new concatenated array or string.

dylanconklin committed:
Implemented string verification, sentiment analysis; Bug fix in messageDB class

Implemented verification of valid strings field (non-empty strings).

Implemented profanity checker during string verification.

Implemented sentiment analysis during string verification.

Fixed bug in messageDB code requiring creation of MessageDB object to run getMessage().

Installed cuss and sentiment packages in npm.

dylanconklin committed:
Implemented checkReply, getThread; Renamed checkReply file to verifyString

Implemented checkReply function, which returns a boolean indicating whether a message has been responded to.

Implemented getThread, which returns a list of messages which make up a feedback/reply thread.

Renamed a file which was inaccurately named.

dylanconklin committed:
Implementation of code generator in progress

Implementing functions to randomly generate a number, verify the availability of the number, and find the next available code if it is not available.

dylanconklin committed:
Fixed part of getNextCode, Added sortChars

Fixed part of getNextCode to handle the last character in the code combination

Added sortChars to ensure string comparisons work with database

dylanconklin committed:
Fixed variable error

imOnVacation committed:
Merge branch Dylan into dev_generate_code

imOnVacation committed:
Merge pull request #13 from PSU-MCECS-SAFE/dev_generate_code

Dev generate code

imOnVacation committed:
removed code that was not in dev branch

dylanconklin committed:
Finished code generation functions; Created Code class

Finished the code generation functions, and wrapped it in a Code class for simplicity. All functions needed are public functions.

The code still needs to be refactored, though. It could be simpler and two functions are remarkably similar, and I will merge them at some point, but it won't affect the way to interact with the class.

Vincentmask committed:
Merge pull request #16 from PSU-MCECS-SAFE/development

for testing

Vincentmask committed:
Modify server to call the genCode to insert message with
 unique code

Vincentmask committed:
Changed the port just to avoid if someone else is using 3001
Catch code back from server to front end

Vincentmask committed:
Modified the syntax of the code
Detail please see the pull request

imOnVacation reviewed (changes requested):
Vincent, can you please merge the updated dev branch into this branch and see if everything working as expected? Thanks!

I also noticed you were not integrated with the verify_string function?

Vincentmask committed:
Merge pull request #21 from PSU-MCECS-SAFE/development

test

imOnVacation committed:
Merge pull request #23 from PSU-MCECS-SAFE/development

Merge dev to Jaafar

imOnVacation committed:
Merge pull request #24 from PSU-MCECS-SAFE/generate_code_and_verify_string

Merge Generate code function to Jaafar

imOnVacation committed:
Revert "Revert "Jaafar""

imOnVacation committed:
Merge branch 'Jaafar' into revert-17-revert-11-Jaafar

imOnVacation committed:
Merge pull request #25 from PSU-MCECS-SAFE/revert-17-revert-11-Jaafar

This Revert will help to get back real Jaafar PR which feature beyond MVP

imOnVacation committed:
made display code work, refresh the page after close button, changed closing msg to modal

Vincentmask committed:
added error handling - when there are error in the backend, submit btn will pop error msg

imOnVacation committed:
removed pop out msg

McJefe reviewed (approved):
The checkProfanities function killed me lol. We should update it at some point, but this all looks good.

Commit:
Updated verify_string to check for Invalid characters

Temporary fix until we can add support for special characters

Commit:
Added isAvailable function to code generator; Bug fixes to code generator; Added submitting current date to database

Adding work done to check for expired codes (not completed).

Bug fixes to properly query database and get results.

Added support to time stamp messages going into database.

Vincentmask committed:
cuss library is not working on commandJS, switch to another one

Vincentmask committed:
added checkString, will alter use if bad word exist, added analysis_result as part of db attibute'

Vincentmask committed:
changed work flow of this file since cuss is not useable

Vincentmask committed:
Website will alert user if bad work exist and not sending to db

Vincentmask committed:
Merge branch 'generate_code_and_verify_string' into pop_out_boxes_and_check_msg_btn

Vincentmask committed:
Merge pull request #29 from PSU-MCECS-SAFE/pop_out_boxes_and_check_msg_btn

Pop out boxes and check msg btn

Vincentmask committed:
Added recieve email endpoint. When receiver wants to recieve code it will send them an eamil and set receive_reply in db to true

Vincentmask committed:
Added a few error handling and fetch when receiver wants to get code email to them

Vincentmask committed:
We don't need this

Vincentmask committed:
Now will alert based on the error message

Vincentmask committed:
Oops wrong port

Vincentmask committed:
more error handling and bug fix, reply_receive flag will be true once user hit yes to receive reply

Vincentmask committed:
bug fixed

Vincentmask committed:
remove the console.log

Commit:
Modified Sentiment Analysis to Use Comparative Score Instead of Point Score

Using the comparative score can be used to determine how strongly a reviewer felt in a future update

Commit:
HUGE Revamp of Code Generator

Major upgrade for the code generator.

Rewrote most of it to make the code make more sense, and reduce complexity.

Fixed bugs relating to database queries.

Added Support for Expiring Codes.

Commit:
Merge branch 'generate_code_and_verify_string' of https://github.com/PSU-MCECS-SAFE/SAFE into generate_code_and_verify_string

Commit:
Fixed Sentiment Analysis Bug

imOnVacation commented:
Please don't touch this PR before meeting with Mark.

imOnVacation committed:
Merge branch 'development' into generate_code_and_verify_string
